hypothesis-python/src/hypothesis/core.py

```diff
@@ -1080,7 +1084,14 @@ def wrapped_test(*arguments, **kwargs):
     return run_test_with_generator
 
 
-def find(specifier, condition, settings=None, random=None, database_key=None):
+def find(
+    specifier,  # type: SearchStrategy
```

Comment:
My view is that - at some point, now and/or later - we will need to change the type hints in a way that makes type-checkers object to previously passing programs.

This could happen because we made a mistake somewhere, something broke because of upstream (the PEP is still provisional!), or we simply changed the type of something.

Would you be content with a section of the docs that explicitly labelled it a *provisional* API, so that type hints can change with minor releases?  We can always stabilize later, but we can't make a stable API provisional!

Reply:
I don't object to making the API provisional given that type hints themselves are, but I think there's a difference between "provisional: We might occasionally break things" and "provisional: We have a known planned change in our future that is going to break literally every single consumer of this feature", and shipping with a non-generic `SearchStrategy` is the latter.

Can you elaborate on why this is difficult? Could we not do something like the following:

1. Add an explicit dependency on the typing backport on Python 2
2. Add a release which makes SearchStrategy subclass Generic without actually shipping type hints
3. *then* do something equivalent to this release but with generic types in the API

I'm sure this is naive and there are all sorts of missing crucial bits, but I'd like to understand what they are!

Reply:
> IMO adding generics in the first pass is simply too much to review in a single pull, because you need a fair breadth of annotations to be sure it's working

So to be clear, my suggestion was to add the generic type and maybe type a handful of methods (e.g. just the `SearchStrategy` public methods) but *not* attempt to type `hypothesis.strategies` as part of that pull request. i.e. add just enough typing to prove the point rather than enough to be particularly useful.

But I'm fine with the suggestion of removing `py.typed` so that these types are non-public and merging this (after review, obviously) if that's your preferred route.